src/vs/workbench/contrib/terminal/browser/terminalDecorationsProvider.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import Severity from 'vs/base/common/severity';
+import { URI } from 'vs/base/common/uri';
+import { localize } from 'vs/nls';
+import { ITerminalService } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { IDecorationData, IDecorationsProvider } from 'vs/workbench/services/decorations/browser/decorations';
+import { Event, Emitter } from 'vs/base/common/event';
+import { Codicon } from 'vs/base/common/codicons';
+import { listErrorForeground, listWarningForeground } from 'vs/platform/theme/common/colorRegistry';
+
+export interface ITerminalDecorationData {
+	tooltip: string,
+	statusIcon: string,
+	color: string
+}
+
+export class TerminalDecorationsProvider implements IDecorationsProvider {
+	readonly label: string = localize('label', "Terminal");
+	private readonly _onDidChange = new Emitter<URI[]>();
+
+	constructor(
+		@ITerminalService private readonly _terminalService: ITerminalService
+	) {
+	}
+
+	get onDidChange(): Event<URI[]> {
+		return this._onDidChange.event;
+	}
+
+	provideDecorations(resource: URI): IDecorationData | undefined {
+		if (resource.scheme !== 'vscode-terminal') {
+			return;
+		}
+		const instance = this._terminalService.getInstanceFromId(parseInt(resource.path));
+		if (!instance) {
```

Comment:
replace with this !instance.statusList.primary && instance.statusList.primary.icon